src/storage/uci/mod.rs

```diff
@@ -0,0 +1,418 @@
+mod uci_config;
+mod uci_option;
+mod uci_section;
+
+use crate::utils::Error;
+use smol::fs::unix::PermissionsExt;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+use std::sync::Mutex;
+use tempfile::NamedTempFile;
+
+use uci_config::UciConfig;
+use uci_option::UciOption;
+use uci_section::UciSection;
+
+use self::uci_option::UciOptionType;
+
+pub struct UciTree {
+    dir: Box<Path>,
+    configs: HashMap<String, UciConfig>,
+    lock: Mutex<bool>,
+}
+
+impl UciTree {
+    pub fn new(root: String) -> Self {
+        Self {
+            dir: Box::from(Path::new(&root)),
+            configs: HashMap::new(),
+            lock: Mutex::new(false),
+        }
+    }
+
+    fn _load_config(&self, name: &str) -> Result<(), Error> {
```

Comment:
crate::error::Result ?

Reply:
Get, I'll change to crate::utils::Result